opentrons-ai-client/src/molecules/ModulesButtonGroup/index.tsx

```diff
@@ -32,10 +33,25 @@ export function ModulesButtonGroup({
                   key={module.type}
                   iconName="plus"
                   onClick={() => {
-                    if (modulesWatch.some(m => m.type === module.type)) {
-                      return
+                    const isTempModule = module.type === TEMPERATURE_MODULE_TYPE
+                    const tempModuleCount = modulesWatch.filter(
+                      m => m.type === TEMPERATURE_MODULE_TYPE
+                    ).length
+
+                    if (isTempModule) {
+                      if (tempModuleCount >= 2) {
+                        return
+                      }
+                    } else {
+                      if (modulesWatch.some(m => m.type === module.type)) {
+                        return
+                      }
+                    }
+                    const moduleWithId = {
+                      ...module,
+                      id: Date.now().toString(),
                     }
```

Comment:
better to use uuid instead of date id

```suggestion
                   import { v4 as uuid } from 'uuid'

                    const moduleWithId = {
                      ...module,
                      id: uuid(),
                    }
```

Reply:
Fixed